assets/js/base/components/state-input/state-input.js

```diff
@@ -30,15 +30,7 @@ const StateInput = ( {
 				name: decodeEntities( countryStates[ key ] ),
 		  } ) )
 		: [];
-	// @todo: remove this code block when issue https://github.com/woocommerce/woocommerce/issues/25854 is merged
-	// Defaults to the first state when selecting a country with states, this is here
-	// until a bug in Woo core is fixed.
-	// see: https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1919
-	useEffect( () => {
-		if ( ! value && options.length ) {
-			onChangeState( options[ 0 ].key );
-		}
-	}, [ country ] );
```

Comment:
One question about removing this, are we going to require WC 4.0.1+ for the blocks? Right now we have 4.0 (does that include patch versions automatically)?

Reply:
The bug was an edge case, I don't think we need to force a version requirement even though it's fixed in a core fix release.